### PR TITLE
add ROIPooling for Fast(er) R-CNN

### PR DESCRIPTION
**Pull request by guoshengCS**
Add ROIPoolLayer for Fast(er) R-CNN

### CODE REVIEW

File: paddle/gserver/layers/ROIPoolLayer.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+class ROIPoolLayer : public Layer {
+protected:
+  size_t channels_;
+  size_t width_;
+  size_t height_;
+  size_t pooledWidth_;
+  size_t pooledHeight_;
+  real spatialScale_;
+
+  MatrixPtr maxIdxs_;
```
Comment (qingqing01) [inline paddle/gserver/layers/ROIPoolLayer.h]: `maxIdxs_` should be int matrix, but there is no int matrix in Paddle. So add comments here use the real matrix instead.

File: python/paddle/trainer/config_parser.py [hunk 2]
```diff
@@ -1732,6 +1732,17 @@ def __init__(self, name, inputs, size, input_num, num_classes,
         self.config.size = size
 
 
+@config_layer('roi_pool')
+class ROIPoolLayer(LayerBase):
+    def __init__(self, name, inputs, pooled_width, pooled_height,
+                 spatial_scale):
```
Comment (qingqing01) [inline python/paddle/trainer/config_parser.py]: `spatial_scale):` -> `spatial_scale, **xargs):`

File: python/paddle/trainer/config_parser.py [hunk 3]
```diff
@@ -1732,6 +1732,17 @@ def __init__(self, name, inputs, size, input_num, num_classes,
...
+        super(ROIPoolLayer, self).__init__(name, 'roi_pool', 0, inputs)
```
Comment (qingqing01) [inline python/paddle/trainer/config_parser.py]: `inputs)` - > `inputs, **xargs):`

File: paddle/gserver/layers/ROIPoolLayer.cpp [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+  size_t batchSize = getInput(0).getBatchSize();
+  size_t numROIs = getInput(1).getBatchSize();
+
+  MatrixPtr dataValue = getInputValue(0);
+  MatrixPtr roiValue = getInputValue(1);
+  resetOutput(numROIs, channels_ * pooledHeight_ * pooledWidth_);
+  MatrixPtr outputValue = getOutputValue();
+
+  if (useGpu_) {
```
Comment (qingqing01) [inline paddle/gserver/layers/ROIPoolLayer.cpp]: Add TODO(xxx): implement on GPU later.

File: paddle/gserver/layers/ROIPoolLayer.cpp [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+        }
+      }
+      batchData += channelOffset;
+      outputData += poolChannelOffset;
+      argmaxData += poolChannelOffset;
+    }
+    bottomROIs += roiOffset;
+  }
+  if (useGpu_) {
+    getOutputValue()->copyFrom(*outputValue);
```
Comment (qingqing01) [inline paddle/gserver/layers/ROIPoolLayer.cpp]: Does the `maxIdxs_`  needs to output?

File: paddle/gserver/layers/ROIPoolLayer.cpp (hunk 4 above)
Comment (guoshengCS) [reply paddle/gserver/layers/ROIPoolLayer.cpp]: Done.

File: paddle/gserver/layers/ROIPoolLayer.h (hunk 1 above)
Comment (guoshengCS) [reply paddle/gserver/layers/ROIPoolLayer.h]: Done.

File: python/paddle/trainer/config_parser.py (hunk 2 above)
Comment (guoshengCS) [reply python/paddle/trainer/config_parser.py]: Done.

File: python/paddle/trainer/config_parser.py (hunk 3 above)
Comment (guoshengCS) [reply python/paddle/trainer/config_parser.py]: Done.

**Pull request merged**